common/loglevel.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+// #ifndef MENDER_LOGLEVEL_HPP
+// #define MENDER_LOGLEVEL_HPP
+
+#include <cassert>
+
+namespace mender::log {
+
+using namespace std;
+
+enum class LogLevel {
+	Fatal = 5,
+	Error = 4,
+	Warning = 3,
+	Info = 2,
+	Debug = 1,
+	Trace = 0,
```

Comment:
AFAICT, the standard POSIX/C log levels go in the opposite order -- the most severe is `0` and `DEBUG` is the highest value. Not sure if it's important here, though.

Reply:
I guess theoretically you could add more log levels after trace, but not typically after fatal. But not important at all, and these values are not stored anywhere anyway.

Reply:
I'll switch it up, just for familiarity.